### PR TITLE
Various cleanup changes

### PR DESCRIPTION
**Pull request by stuwilkins**

### CODE REVIEW

File: src/imagemodule.c [hunk 1]
```diff
@@ -127,8 +132,15 @@ static PyObject* image_stackmean(PyObject *self, PyObject *args){
     goto error;
   }
   
-  if(stackmean((data_t*)PyArray_DATA(input), (data_t*)PyArray_DATA(mout),
-               (long int*)PyArray_DATA(nout), ndims, dims, norm)){
+  // Ok now we don't touch Python Object ... Release the GIL
+  Py_BEGIN_ALLOW_THREADS
+  
+  int retval = stackmean((data_t*)PyArray_DATA(input), (data_t*)PyArray_DATA(mout),
```
Comment (tacaswell) [inline src/imagemodule.c]: do you need to grab and save these pointers out side of the allow threads block?

Comment (stuwilkins) [inline src/imagemodule.c]: Actually I don't know ...

Comment (stuwilkins) [inline src/imagemodule.c]: I would have thought that the only race condition here is that after the GIL is released the object is changed such that the pointer changes.

Comment (tacaswell) [inline src/imagemodule.c]: The other thing to be careful of is the reference counting and I do not know what `PyArray_DATA` does in that regard.

Comment (stuwilkins) [inline src/imagemodule.c]: From bitter experience there is no reference counting there .....

**Pull request merged**